### PR TITLE
Cppcheck fixes

### PR DESCRIPTION
**Pull request by neusdan**

### CODE REVIEW

File: src/lib/entropy/win32_stats/es_win32.cpp [hunk 1]
```diff
@@ -72,7 +72,6 @@ void Win32_EntropySource::poll(Entropy_Accumulator& accum)
 
    if(!accum.polling_finished())
       {
-      size_t heap_lists_found = 0;
```
Comment (randombit) [inline src/lib/entropy/win32_stats/es_win32.cpp]: This move changes the semantics of the poll I think. When here, the variable represented the total calls  to Heap32ListFirst, after move limits total calls to Heap32First.

File: src/lib/prov/tpm/tpm.cpp [hunk 2]
```diff
@@ -388,7 +388,6 @@ class TPM_Signing_Operation : public PK_Ops::Signature
          {
          if(auto* key = dynamic_cast<const TPM_PrivateKey*>(&spec.key()))
             {
-            const std::string padding = spec.padding();
```
Comment (randombit) [inline src/lib/prov/tpm/tpm.cpp]: This shouldn't be removed, actually there is a missing check on the value - TPM can only sign with PKCS v1.5 padding so for example if padding was PSS this should throw as unsupported.

**Comment by randombit** [comment]
Looks good, thanks. Can you squash this down? Ideally into one commit with the `explicit` additions and one with everything else but the changes are somewhat mixed so a single commit with everything is ok too.

I'm somewhat indifferent on the semantic change to the Win32 stats caused by the variable move; it results in more heap infos being hashed in, which maybe is good (more bits), maybe is bad (slower).

File: src/lib/entropy/win32_stats/es_win32.cpp (hunk 1 above)
Comment (neusdan) [inline src/lib/entropy/win32_stats/es_win32.cpp]: Are you sure? I can't see any loop around `Heap32ListFirst()`

**Comment by neusdan** [comment]
Thanks for the review. Should i force push with the squashed commits or make a new pull request?

**Comment by neusdan** [comment]
Made a new PR with the explicit changes and will make another one when these changes are merged.

**Comment by randombit** [comment]
Squash and force push to the existing branch is fine by me.

File: src/lib/entropy/win32_stats/es_win32.cpp (hunk 1 above)
Comment (randombit) [inline src/lib/entropy/win32_stats/es_win32.cpp]: You're right I totally misread this, n/m

**Comment by neusdan** [comment]
Closed in favor of #444 and #445

**Pull request closed without merging**